text/0000-feature-metadata.md

```diff
@@ -0,0 +1,275 @@
+             + unicode          Enables all Unicode features
+             - unstable (U)     Some unstable features
+             - deprecated (D)   Not a real feature, but it could be
+
+    Updating crates.io index
+```
+
+Features like `aho-corasick`, `memchr`, or `use_std` would likely be `public =
+false` since they aren't listed on the crate landing page.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+-   Added complexity to Cargo. Parsing is trivial, but exact implementation
+    details do add test surface area
+-   Added Cargo arguments if escape hatches for `public` or `unstable` are created
+-   Docstrings can be lengthy, adding noise to `Cargo.toml`. This could
+    potentially be solved with the below mentioned `doc-file` key.
+-   `unstable` and `public` uses may not be common enough to be worth including
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+-   TOML-docstrings (`## Some doc comment`) and attributes that mimic Rust
+    docstrings could be used instead of a `doc` key. This is discussed in
+    [future-possibilities], but decided against to start since it is much simpler
+    to parse standard TOML.
+-   Feature descriptions could be specified somewhere in Rust source files. This
+    has the downside of creating multiple sources of truth on features.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+-   There is an existing crate that uses TOML comments to create a features table:
+    <https://docs.rs/document-features/latest/document_features/>
+-   `docs.rs` displays a feature table, but it is fairly limited
+-   Ivy has a [visibility attribute] for its configuration (mentioned in [cargo #10882])
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+-   Should `foo = {}` be allowed, or is `foo = { requires = [] }` the minimum if
+    only using this field? I am leaning toward having it required to avoid
+    inconsistency like
+
+    ```toml
+    foo = {}
+    bar = []
+    baz = {}
+    ```
+
+    However, I would also say that `foo = { deprecated = true }` (no `requires`
+    key) could be valid.
+
+-   Do we want all the proposed keys? Specifically, `public` and `unstable` may be
+    more than what is needed.
+
+    See also:
+
+    -   <https://github.com/rust-lang/cargo/issues/10882>
+    -   <https://github.com/rust-lang/cargo/issues/10881>
+
+-   If we use the semantics as-written, should there be a
+    `--allow-private-features` or `--allow-unstable-features` flag? Or how should
+    a user opt in?
+-   Does it make sense to have separate `hidden` (not documented) and `public`
+    (feature not allowed downstream) attribute? I think probably not
+-   Should there be a way to deny deprecated features?
+
+It is worth noting that simpler keys (`requires`, `doc`, `deprecated`) could be
+stabilized immediately and other features could be postponed.
+
+# Future possibilities
+
+[future-possibilities]: #future-possibilities
```

Comment:
- `cargo add` showing the `doc` key
- A [cargo info](https://github.com/rust-lang/cargo/issues/948) to show the `doc` key and other context

Reply:
I have these listed now